scripts/_vault_commands.py

```
@@ -0,0 +1,23 @@
+
+def _generate_secret_id(payload: str) -> str:
+    """Extract a secret-id from a Vault response payload."""
+    try:
+        data: Dict[str, Any] = json.loads(payload)
+    except json.JSONDecodeError as exc:  # pragma: no cover - defensive guard
+        msg = "Failed to decode Vault response whilst generating a secret-id"
+        raise VaultBootstrapError(msg) from exc
+
+    if secret_id := data.get("data", {}).get("secret_id"):
+        return secret_id
+    raise VaultBootstrapError("Vault response missing secret_id field")
```

Ruff (0.14.1)

Avoid specifying long messages outside the exception class

(TRY003)